src/realm-settings/PartialImport.tsx

```diff
@@ -278,107 +285,258 @@ export const PartialImportDialog = (props: PartialImportProps) => {
+                      "clientRoles",
+                      t("common:clientRoles")
+                    )}
+                </DataList>
+              </StackItem>
+              <StackItem>
+                <Text>{t("selectIfResourceExists")}:</Text>
                 <Select
-                  toggleId="realm-selector"
-                  isOpen={isRealmSelectOpen}
-                  onToggle={() => setIsRealmSelectOpen(!isRealmSelectOpen)}
-                  onSelect={handleRealmSelect}
-                  placeholderText={targetRealm.realm || targetRealm.id}
+                  isOpen={isCollisionSelectOpen}
+                  direction="up"
+                  onToggle={() => {
+                    setIsCollisionSelectOpen(!isCollisionSelectOpen);
+                  }}
+                  onSelect={handleCollisionSelect}
+                  placeholderText={t(collisionOption)}
                 >
-                  {realmSelectOptions()}
+                  {collisionOptions()}
                 </Select>
               </StackItem>
-            )}
-            <StackItem>
-              <Text>{t("chooseResources")}:</Text>
-              <DataList aria-label="Resources to import" isCompact>
-                {targetHasResource("users") &&
-                  resourceDataListItem("users", "users")}
-                {targetHasResource("groups") &&
-                  resourceDataListItem("groups", "groups")}
-                {targetHasResource("clients") &&
-                  resourceDataListItem("clients", "clients")}
-                {targetHasResource("identityProviders") &&
-                  resourceDataListItem(
-                    "identityProviders",
-                    "identity providers"
-                  )}
-                {targetHasRealmRoles() &&
-                  resourceDataListItem("realmRoles", "realm roles")}
-                {targetHasClientRoles() &&
-                  resourceDataListItem("clientRoles", "client roles")}
-              </DataList>
-            </StackItem>
-            <StackItem>
-              <Text>{t("selectIfResourceExists")}:</Text>
-              <Select
-                isOpen={isCollisionSelectOpen}
-                direction="up"
-                onToggle={() => {
-                  setIsCollisionSelectOpen(!isCollisionSelectOpen);
-                }}
-                onSelect={handleCollisionSelect}
-                placeholderText={t(collisionOption)}
-              >
-                {collisionOptions()}
-              </Select>
-            </StackItem>
-          </>
-        )}
-      </Stack>
-    </Modal>
+            </>
+          )}
+        </Stack>
+      </Modal>
+    );
+  };
+
+  const importCompleteMessage = () => {
+    return t("importSuccess", {
+      added: importResponse?.added,
+      overwritten: importResponse?.overwritten,
+      skipped: importResponse?.skipped,
+    });
+  };
+
+  const loader = async (first?: number, max?: number) => {
```

Comment:
You can reduce the amount of non-null assertions and optional chaining here by using an early return on the presence of `importResponse` (like already at the end of this function).

```suggestion
  const loader = async (first?: number, max?: number) => {
    // Empty state can't happen as you will get back a result record
    // for each record you submit. This is here to make Typescript happy.
    if (!importResponse) {
      return [];
    }
```